### PR TITLE
restructure output results; reduce logging spam

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import java.time.Instant;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class Meta {
+  String request;
+  Instant time;
+  Integer status;
```
Comment (jhulbert-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java:12]: Should these be more explicit?
`requestBody`, `httpCode`

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java [hunk 2]
```diff
@@ -46,26 +51,49 @@ public void create(Resource resource) {
...
+    String testResultsPath = propertiesFileLocation + "/results/" + name;
+    if (new File(testResultsPath).exists()) {
+      FileSystemUtils.deleteRecursively(new File(testResultsPath));
+    }
+    new File(testResultsPath).mkdir();
+
+    new File(testResultsPath + "/response.json").createNewFile();
+    FileWriter fw = new FileWriter(testResultsPath + "/response.json", StandardCharsets.UTF_8);
     fw.write(response.prettyPrint());
     fw.close();
```
Comment (jhulbert-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java:74]: This is repeated thrice.

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java (hunk 1 above)
Comment (bschofield-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java:12]: Favor `int` over `Integer`

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java [hunk 3]
```diff
@@ -46,26 +51,49 @@ public void create(Resource resource) {
...
-  private void saveResultsToDisk(Response response) {
-    new File(propertiesFileLocation + "/responses").mkdir();
-    new File(propertiesFileLocation + "/responses/" + name + ".response").createNewFile();
-    FileWriter fw =
-        new FileWriter(
-            propertiesFileLocation + "/responses/" + name + ".response_" + Instant.now(),
-            StandardCharsets.UTF_8);
-    fw.write(String.format("%s%n", response.statusCode()));
+  private void saveResultsToDisk(URL url, Response response, Optional<Resource> body) {
+    new File(propertiesFileLocation + "/results").mkdir();
```
Comment (bschofield-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java:63]: `mkdirs()` will make _any_ directory in the path

This assume `/` is a directory separator, which is not always the case. Take advantage of `File` constructors
```var resultsDir = new File(propertiesFileLocation,"results")```

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java [hunk 4]
```diff
@@ -46,26 +51,49 @@ public void create(Resource resource) {
...
+
+    String testResultsPath = propertiesFileLocation + "/results/" + name;
+    if (new File(testResultsPath).exists()) {
+      FileSystemUtils.deleteRecursively(new File(testResultsPath));
+    }
+    new File(testResultsPath).mkdir();
+
+    new File(testResultsPath + "/response.json").createNewFile();
```
Comment (bschofield-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java:71]: This is weird to do file naming string concat thing twice. `FileWriter` takes a `File`
```
var testResultsDir = new File(resultsDir,name);
var responseFile = new File(testResultsDir,"response.json");
var write = new FileWriter(responseFile, UTF_8);
```
```

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java (hunk 2 above)
Comment (bschofield-va) [reply vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/InteractiveTestContext.java:74]: And it should be using `try-with-resources` to ensure the handles are closed.

File: vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package gov.va.api.health.vistafhirquery.interactivetests;
+
+import java.time.Instant;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class Meta {
```
Comment (bschofield-va) [inline vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java:9]: `Meta` looks like R4 `Meta` and DSTU2 `Meta` and STU3 `Meta`. So much `Meta`

Comment (ilaflamme-va) [reply vista-fhir-query-interactive-tests/src/main/java/gov/va/api/health/vistafhirquery/interactivetests/Meta.java]: `TestMeta`, `RéquestMeta`, `TestMetadata` ... preference?

**Review by jhulbert-va — APPROVED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**